### PR TITLE
fix(macros/jsxref): reference to same page in some cases

### PR DESCRIPTION
**Pull request by yin1999**
When referenced l10n page does not exist but actually in en-US, the URL
may lead to a different one than the one in en-US.

## Summary

Fixes: #6671.

### Problem

see the above issue.

### Solution

get page in content repo only, so the same macro call will lead to the same document (may or may not be localized).

## Screenshots

check out mdn/translated-content#6707. go to URL: `http://localhost:5042/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Error`

### Before

image

### After

image

## How did you test this change?

check out mdn/translated-content#6707 and then run dev server.

test in browser.

### CODE REVIEW

File: kumascript/tests/macros/jsxref.test.js [hunk 1]
```diff
@@ -163,6 +163,7 @@ describeMacro("jsxref", function () {
 
       macro.ctx.info.getPageByURL = jest.fn((url) => {
```
Comment (fiji-flo) [inline kumascript/tests/macros/jsxref.test.js:164]: I think to make the test correct we have to add
```js
      const default_ref_url = js_ref_url + partial_slug;
      const default_glob_url = `${js_ref_url}Global_Objects/${partial_slug}`;
```
just above and replace the mock with this:
```js
      macro.ctx.info.getPageByURL = jest.fn((url) => {
        url = getPathname(url);
        if (url === glob_url) {
          return {
            url: glob_url,
            slug: js_ref_slug + partial_slug,
          };
        } else if (url === default_glob_url) {
          return {
            url: default_glob_url,
            slug: js_ref_slug + partial_slug,
          };
        } else if (url === default_ref_url || url === ref_url) {
          return {};
        }
      });
```

**Review by fiji-flo — CHANGES_REQUESTED** [review]

File: kumascript/tests/macros/jsxref.test.js (hunk 1 above)
Comment (yin1999) [reply kumascript/tests/macros/jsxref.test.js]: Thanks for suggestions. I've update the testing case and rebased

**Comment by yin1999** [comment]
Hi @fiji-flo. Is there anything else to be aware of?

**Comment by caugner** [comment]
@yin1999 I had a hard time understanding exactly what the macro does, so I restructured it in f9e747e2e90e1e2b52c7ed73a0fa5d54b575526c. Could you please check if the new structure makes sense to you?

**Review by yin1999 — COMMENTED** [review]
Hi @caugner. This really make sense. I have only one small question

File: kumascript/macros/jsxref.ejs [hunk 2]
```diff
@@ -9,45 +9,60 @@
...
-const defaultURL = `/en-US/docs/Web/JavaScript/Reference/`
-const page = wiki.getPage(defaultURL + $0);
-const objectPage = wiki.getPage(defaultURL + slug + '/' + $0);
+let slug;
+if (apiName.includes(".") && !apiName.includes("..")) {
+    const normalizedApiName = apiName.replace('()', '').replace('.prototype.', '.');
+    // E.g. "Array.filter", but not "try...catch".
+    slug = normalizedApiName.replace('.', '/');
+} else {
+    slug = apiName;
```
[the remark this answers is not in the document]
Comment (caugner) [reply kumascript/macros/jsxref.ejs]: Good catch, that makes a lot of sense, and makes it even easier than that, see: 3f5ed38f54e45b43dea141d51645b01f1f2f86f3

[the remark this answers is not in the document]
Comment (yin1999) [reply kumascript/macros/jsxref.ejs]: LGTM. Thanks @caugner.

**Review by caugner — APPROVED** [review]

**Comment by caugner** [comment]
Thank you @yin1999! 🎉

**Pull request merged**